src/torrent.c

```diff
@@ -485,6 +485,14 @@ gint64 torrent_get_queue_position(JsonObject *args)
         return -1;
 }
 
+JsonArray *torrent_get_labels(JsonObject *t)
+{
+    if (json_object_has_member(t, FIELD_LABELS))
+        return json_object_get_array_member(t, FIELD_LABELS);
```

Comment:
This isn't a new problem in this file but this doesn't actually validate the type of the data is correct.

To quote the docs:

> It is a programmer error to call this on a node which doesn’t hold an array value. Use JSON_NODE_HOLDS_ARRAY first.

Reply:
I wrote these simple wrappers to safely get various types, it may be helpful: https://gitlab.gnome.org/GNOME/epiphany/-/blob/master/src/webextension/ephy-json-utils.h

Reply:
I was planning on doing something like that but I'd prefer to keep this at GPLv2+ for now and not GPLv3+ which those are licensed at. So for now I won't look at their implementation. As an API it looks good though .

Reply:
Since I am the sole author of that file I will relicense it GPLv2+ :)

Reply:
Works for me